Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp

```diff
@@ -332,7 +338,22 @@ void WebFoundTextRangeController::setTextIndicatorWithRange(const WebCore::Simpl
     frame->selection().setUpdateAppearanceEnabled(false);
 #endif
 
-    m_textIndicator = WebCore::TextIndicator::createWithRange(range, options, WebCore::TextIndicatorPresentationTransition::None, WebCore::FloatSize(indicatorMargin, indicatorMargin));
+    return WebCore::TextIndicator::createWithRange(range, options, transition, WebCore::FloatSize(indicatorMargin, indicatorMargin));
+}
+
+void WebFoundTextRangeController::setTextIndicatorWithRange(const WebCore::SimpleRange& range)
+{
+    m_textIndicator = createTextIndicatorForRange(range, WebCore::TextIndicatorPresentationTransition::None);
+}
+
+void WebFoundTextRangeController::flashTextIndicatorAndUpdateSelectionWithRange(const WebCore::SimpleRange& range)
+{
+    auto& document = range.startContainer().document();
+    document.selection().setSelection(VisibleSelection(range), FrameSelection::defaultSetSelectionOptions(UserTriggered));
+
+    auto textIndicator = createTextIndicatorForRange(range, WebCore::TextIndicatorPresentationTransition::Bounce);
+    if (textIndicator)
+        m_webPage->corePage()->chrome().client().setTextIndicator(textIndicator->data());
```

Comment:
Nit - `if (auto textIndicator = …) {`

Since we're in WebKit2 code already, perhaps it _might_ also be simpler to just call `WebPage::send()` or a new `WebPage::setTextIndicator()` helper method directly, instead of bouncing to `Page` and back?

Reply:
Added a new helper.